bundles/org.eclipse.sirius.emfjson/src/main/java/org/eclipse/sirius/emfjson/utils/GsonEObjectDeserializer.java

```diff
@@ -540,15 +542,13 @@ private void deserializeSingleNonContainmentEReference(EReference eReference, Js
                     this.helper.setValue(eObject, eReference, object);
                 }
             } else {
-                String resourceURIPath = id.substring(0, index); // The URI of the resource where I can find
-                // the EObject
-                EPackage ePackage = this.getPackageForURI(resourceURIPath); // return an EPackage only if the
-                // URI represent an EPackage
-                Resource packageResource = null;
+                String resourceURIPath = id.substring(0, index);
                 EObject object = null;
-                if (ePackage != null) {
-                    packageResource = ePackage.eResource();
-                    object = packageResource.getEObject(fragmentEMF);
+                if (resourceURIPath.indexOf(SCHEME_SEPARATOR) != -1) {
+                    EPackage ePackage = this.getPackageForURI(resourceURIPath);
+                    if (ePackage != null) {
+                        object = ePackage.eResource().getEObject(fragmentEMF);
+                    }
                 } else {
                     object = this.createProxyEObject(id, qualifiedType, eReference);
                 }
```

Comment:
I think the new logic is wrong. It assumes only resources from the package registry have full URI paths (with scheme, separator, etc.). This will not be the case in e.g. SysON where standard libraries, which are plain models, have URIs of the form `kermllibrary://<uuid>`. With the new code, such a URI would be lookup up in the package registry (because is has a `:`), which will fail, and we'll stop there. We should fall back to the `this.createProxyEObject()` in this case as in a non-URI-like path.